src/libraries/System.IO.Ports/Directory.Build.props

```diff
@@ -3,10 +3,10 @@
   <PropertyGroup>
     <StrongNameKeyId>Open</StrongNameKeyId>
     <IncludePlatformAttributes>true</IncludePlatformAttributes>
-    <UnsupportedOSPlatforms>browser</UnsupportedOSPlatforms>
+    <UnsupportedOSPlatforms>browser;android;ios;tvos</UnsupportedOSPlatforms>
```

Comment:
Are using `UnsupportedOSPlatforms` attribute because the suggested support list (Windows, Linux, OSX, FreeBSD) might not accurate? Even if you added all known unsupported platforms now in the future you will need to add all new unsupported platforms. If this project certainly only supported on (Windows, Linux, OSX, FreeBSD)  I would prefer to use `SupportedOSPlaform("windows"), SupportedOSPlaform("Linux"), SupportedOSPlaform("OSX"), SupportedOSPlaform("FreeBSD")` attributes instead

Reply:
What's the most convenient way to apply assembly-level `[SupportedOSPlatform]` attributes?  I don't think we ever finished implementing a `<SupportedOSPlatforms>` MSBuild property within our targets.

Reply:
I agree that using an inclusive list would be better but I couldn't find a way how to configure it with the current build

Reply:
I drafted that capability out a while back in #41209, but we didn't actually _need_ it at that time, so I didn't push it through. I don't think we should hold this PR up for that, but I'd be happy to resurrect that PR. I'll reopen it and then after this PR merges, I'll update `System.IO.Ports` to use the inclusive list.

Reply:
> What's the most convenient way to apply assembly-level `[SupportedOSPlatform]` attributes

Probably better to  apply the same as how we apply `[UnsupportedOSPlatform]` attribute

> I agree that using an inclusive list would be better but I couldn't find a way how to configure it with the current build

Now i am in doubt, we already have an [UnsupportedOSPlatform("browser")] attribute, and if we add `SupportedOSPlaform("windows"), SupportedOSPlaform("Linux"), SupportedOSPlaform("OSX"), SupportedOSPlaform("FreeBSD")` it is causing inconsistent scenario, most cases that will be OK but could cause the analyzer misbehave. So if we add the Supported attributes we need to remove the `[UnsupportedOSPlatform("browser")]` (or update the analyzer to ignore unsupported (Deny list) in case there is Allow list).  So my question is could we remove the  `[UnsupportedOSPlatform("browser")]` in case we make it an inclusive list? Or I can make a quick update to the analyzer, anyway, as we could not apply an inclusive list right now I am OK with this PR as is

Reply:
>  So if we add the Supported attributes we need to remove the [UnsupportedOSPlatform("browser")]

That makes sense to me